Adding check for a single batch in adaptive_avg_pool

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23137 Adding check for a single batch in adaptive_avg_pool**

Differential Revision: [D16403804](https://our.internmc.facebook.com/intern/diff/D16403804)

Commit:
Adding check for a single batch in adaptive_avg_pool

zafartahirov commented:
Inner part of nested `parallel_for`'s are treated as serial for loops. If there is a single batch, we can treat it as if the rank of the input was 3.

Commit:
Update on "Adding check for a single batch in adaptive_avg_pool"

Adding check for a single batch in adaptive_avg_pool

gh-metadata: pytorch pytorch 23137 gh/zafartahirov/11/head

Commit:
Update on "Adding check for a single batch in adaptive_avg_pool"

Adding check for a single batch in adaptive_avg_pool

gh-metadata: pytorch pytorch 23137 gh/zafartahirov/11/head

Commit:
Update on "Adding check for a single batch in adaptive_avg_pool"

Adding check for a single batch in adaptive_avg_pool

gh-metadata: pytorch pytorch 23137 gh/zafartahirov/11/head

Commit:
Update on "Adding check for a single batch in adaptive_avg_pool"

Adding check for a single batch in adaptive_avg_pool

gh-metadata: pytorch pytorch 23137 gh/zafartahirov/11/head
